plugins/source/postgresql/client/sync.go

```diff
@@ -141,7 +142,17 @@ func syncTable(ctx context.Context, tx pgx.Tx, table *schema.Table, res chan<- m
 
 			scalar.AppendToBuilder(builder.Field(i), s)
 		}
-		res <- &message.SyncInsert{Record: builder.NewRecord()} // NewRecord resets the builder for reuse
+
+		processed++
+		if processed%c.pluginSpec.RowsPerRecord == 0 {
+			res <- &message.SyncInsert{Record: builder.NewRecord()} // NewRecord resets the builder for reuse
+			processed = 0
+		}
+	}
+
+	record := builder.NewRecord()
+	if record.NumRows() > 0 { // only send if there are some unsent rows
```

Comment:
I think it's safe, but it's been a while since I've worked with this so just checking: is it safe to call `builder.NewRecord` on a builder with 0 rows?

Reply:
Yeah no problem there